Extension/test/scenarios/SingleRootProject/tests/lmTool.test.ts

```diff
@@ -0,0 +1,206 @@
+            throw new Error('Method not implemented.');
+        }
+        validatePosition(_position: vscode.Position): vscode.Position {
+            throw new Error('Method not implemented.');
+        }
+    }
+    beforeEach(() => {
+        proxyquire.noPreserveCache(); // Tells proxyquire to not fetch the module from cache
+        // Ensures that each test has a freshly loaded instance of moduleUnderTest
+        moduleUnderTest = proxyquire(
+            '../../../../src/LanguageServer/lmTool',
+            {} // Stub if you need to, or keep the object empty
+        );
+
+        sinon.stub(util, 'extensionContext').value({ extension: { id: 'test-extension-id' } });
+
+        mockTextDocumentStub = sinon.createStubInstance(MockTextDocument);
+        mockTextEditorStub = new MockTextEditor(new vscode.Selection(0, 0, 0, 0), [], [], { tabSize: 4 }, mockTextDocumentStub);
+        mockLanguageModelToolInvocationOptions = new MockLanguageModelToolInvocationOptions();
+        activeClientStub = sinon.createStubInstance(DefaultClient);
+        const clientsStub = sinon.createStubInstance(ClientCollection);
+        sinon.stub(extension, 'getClients').returns(clientsStub);
+        sinon.stub(clientsStub, 'ActiveClient').get(() => activeClientStub);
+        activeClientStub.getIncludes.resolves({ includedFiles: [] });
+        sinon.stub(vscode.window, 'activeTextEditor').get(() => mockTextEditorStub);
+        logLanguageModelToolEventStub = sinon.stub(telemetry, 'logLanguageModelToolEvent').returns();
+    });
+
+    afterEach(() => {
+        sinon.restore();
+    });
+
+    const arrange = ({ chatContext, requestedContentTypes }:
+    { chatContext?: ChatContextResult; requestedContentTypes: string[] } =
+    { chatContext: undefined, requestedContentTypes: [] }
+    ) => {
+        activeClientStub.getChatContext.resolves(chatContext);
+        mockLanguageModelToolInvocationOptions.requestedContentTypes = requestedContentTypes;
+        sinon.stub(util, 'isCpp').returns(true);
+        sinon.stub(util, 'isHeaderFile').returns(false);
+    };
+
+    it('should provide cpp context.', async () => {
+        arrange({
+            requestedContentTypes: ['text/plain'],
+            chatContext: {
+                language: 'c++',
+                standardVersion: 'c++20',
+                compiler: 'msvc',
+                targetPlatform: 'windows',
+                targetArchitecture: 'x64'
+            }
+        });
+
+        const result = await new moduleUnderTest.CppConfigurationLanguageModelTool().invoke(mockLanguageModelToolInvocationOptions, new vscode.CancellationTokenSource().token);
+
+        ok(result['text/plain'], 'result should contain a text/plain entry');
+        ok(result['text/plain'] === 'The user is working on a c++ project. The project uses language version C++20, compiles using the MSVC compiler, targets the Windows platform, and targets the x64 architecture.');
```

Comment:
I think `c++` should be `C++`

Reply:
Oh, but this is just a test? Can the source that uses `c++` be changed?

Reply:
Copilot says:
![image](https://github.com/user-attachments/assets/21e25ccd-aedf-4a0e-8db8-537428ff4bc6)
so it seems better to use prompts it thinks are correct.

Reply:
Thanks, fixed.